### PR TITLE
Migrations: migrate admin user create page

### PR DESCRIPTION
**Pull request by dprokop**
Ref  #21484 

TODO:
- [x] Remove unused code from AdminEditUserCtrl and the template file
- [x] Add validation to the form

### Notes for reviewers

In this PR I have added [react-hooks-form](https://react-hook-form.com/get-started) lib to grafana/ui to handle forms state/validation. It's re-exported from grafana/ui for now, but I would rather like to hide this library behind `<Form />` component that would handle forms logic - this isn't a scope of this PR though. 
Here's a nice overview of react-hooks-form https://blog.logrocket.com/react-hook-form-vs-formik-a-technical-and-performance-comparison/ that I found very informative

EDIT:
Hidden form hook behind Form component

### CODE REVIEW

File: public/app/features/admin/UserCreatePage.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+interface UserDTO {
+  name: string;
+  password: string;
+  email?: string;
+  login?: string;
+}
+
+const createUser = async (user: UserDTO) => getBackendSrv().post('/api/admin/users', user);
```
Comment (mckn) [inline public/app/features/admin/UserCreatePage.tsx]: If this request fails, how do we handle the errors? Do we have any globally defined error handling?

**Review by mckn — APPROVED** [review]
Looks really good! Awesome work 🙌

File: public/app/features/admin/UserCreatePage.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/admin/UserCreatePage.tsx]: Yes, an alert will be shown

File: public/app/features/admin/UserCreatePage.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                  required
+                  invalid={!!errors.password}
+                  error={!!errors.password && 'Password is required and must contain at least 4 characters'}
+                >
+                  <Forms.Input
+                    size="md"
+                    type="password"
+                    name="password"
+                    ref={register({
+                      validate: value => {
```
Comment (mckn) [inline public/app/features/admin/UserCreatePage.tsx]: nit: this could be a one-liner, right?

**Review by mckn — APPROVED** [review]
Looks really good! I really like the `react-hook-form` addition 🙌

**Comment by torkelo** [comment]
Great job! 

Really happy with how this is turning out. 

For me the input background is bugging, swithes to black randomly some times. Anyone else seen this? 
Screen Shot 2020-01-16 at 14 21 01

File: public/app/routes/routes.ts [hunk 3]
```diff
@@ -302,13 +302,12 @@ export function setupAngularRoutes($routeProvider: route.IRouteProvider, $locati
...
+      template: '<react-container />',
+      resolve: {
+        component: () =>
+          SafeDynamicImport(import(/* webpackChunkName: "UserCreatePage" */ 'app/features/admin/UserCreatePage')),
+      },
     })
-    // .when('/admin/users/edit/:id', {
-    //   templateUrl: 'public/app/features/admin/partials/edit_user.html',
-    //   controller: 'AdminEditUserCtrl',
-    // })
```
Comment (torkelo) [inline public/app/routes/routes.ts]: should this be removed?

**Review by torkelo — APPROVED** [review]
❤️

File: public/app/routes/routes.ts (hunk 3 above)
Comment (dprokop) [reply public/app/routes/routes.ts]: yes

**Comment by dprokop** [comment]
> For me the input background is bugging, swithes to black randomly some times. Anyone else seen this?

@torkelo that's because of chrome's default styles applied to autofilled inputs. It's updated now.

**Review by torkelo — APPROVED** [review]
Great!

**Pull request merged**